File: tkpdflight/src/main/java/com/tokopedia/flight/flightsearch/domain/FlightSearchUseCase.java
package com.tokopedia.flight.flightsearch.domain;

import com.tokopedia.flight.airport.data.source.db.model.FlightAirportDB;
import com.tokopedia.flight.common.domain.FlightRepository;
import com.tokopedia.flight.flightsearch.data.db.model.FlightSearchSingleRouteDB;
import com.tokopedia.flight.flightsearch.data.repository.FlightSearchRepository;
import com.tokopedia.usecase.RequestParams;
import com.tokopedia.usecase.UseCase;

import java.util.List;

import javax.inject.Inject;

import rx.Observable;

/**
 * Created by zulfikarrahman on 10/25/17.
 */

public class FlightSearchUseCase extends UseCase<List<FlightSearchSingleRouteDB>> {
    private final FlightSearchRepository flightSearchRepository;

    @Inject
    public FlightSearchUseCase(FlightSearchRepository flightSearchRepository) {
        this.flightSearchRepository = flightSearchRepository;
    }

    //TODO request params for usecase
    @Override
    public Observable<List<FlightSearchSingleRouteDB>> createObservable(RequestParams requestParams) {
        return flightSearchRepository.getFlightSearch(false);
    }

}

File: features/digital/flight/src/main/java/com/tokopedia/flight/searchV2/domain/FlightSearchUseCase.kt
package com.tokopedia.flight.searchV2.domain

import com.tokopedia.flight.searchV2.data.repository.FlightSearchRepository
import com.tokopedia.flight.searchV2.presentation.model.FlightSearchApiRequestModel
import com.tokopedia.flight.searchV2.presentation.model.FlightSearchMetaViewModel
import com.tokopedia.usecase.RequestParams
import com.tokopedia.usecase.UseCase
import rx.Observable

/**
 * Created by Rizky on 26/09/18.
 */
class FlightSearchUseCase(val flightSearchRepository: FlightSearchRepository) : UseCase<FlightSearchMetaViewModel>() {

    override fun createObservable(requestParams: RequestParams): Observable<FlightSearchMetaViewModel>? {
        return flightSearchRepository.getSearchSingle(requestParams.parameters)
    }

    fun createReqestParams(flightSearchApiRequestModel: FlightSearchApiRequestModel) : RequestParams {
        val requestParams = RequestParams.create()
        return requestParams
    }

}